### DISCUSSION TITLE
tree.edit() for highlighting

### DISCUSSION THREAD

**Discussion by maxwell8888** [tree-sitter/tree-sitter#1530] [Q&A]
We can make edits to a tree like this:
```rust
let tree = parser.parse(source_code, None).unwrap();
tree.edit(InputEdit {
  start_byte: 8,
  old_end_byte: 8,
  new_end_byte: 14,
  start_position: Point::new(0, 8),
  old_end_position: Point::new(0, 8),
  new_end_position: Point::new(0, 14),
});
```
However, the highlighting API doesn't seem to allow making edits to the tree like above. I just takes a source code string, parses it, runs the highlight query and returns the results of the query, but doesn't return the parsed tree so there is no way to make an edit.  
```rust
let mut highlight = Highlighter::new();
let events = highlight
    .highlight(&highlight_config, source_code, None, |_| None)
    .unwrap();
```

It seems like if I want to make edits to a tree used for highlighting, I should just ignore the tree-sitter-highlight crate and just parse the code and run the highlight queries using the querying API provided by the tree-sitter crate. In either case, please could somebody provide some guidance on what the best way to go about incrementally updating highlighted code is, i.e. for highlighting in a text editor.

**Comment by markusstephanides** [comment]
Hi! Just wanted to ask if there is a solution for this?
